src/Microsoft.Extensions.Caching.Memory/MemoryCacheOptions.cs

```diff
@@ -13,6 +13,12 @@ public class MemoryCacheOptions : IOptions<MemoryCacheOptions>
 
         public TimeSpan ExpirationScanFrequency { get; set; } = TimeSpan.FromMinutes(1);
 
+        public bool EnforceSizeLimit { get; set; } = false;
+
+        public long HighWatermark { get; set; }
```

Comment:
Naming, docs.
Make it nullable rather than having a separate EnforceSizeLimit bool?